fornax-hea/build-sas.sh

```diff
@@ -0,0 +1,194 @@
+  echo "Error: No matching files found for '$pattern'." >&2
+  exit 1
+elif [[ $num_matches -gt 1 ]]; then
+  echo "Error: Multiple matching files found for '$pattern'." >&2
+  exit 1
+else
+  # A single, real match was found, so we split the file name to retrieve the installation folder
+  sas_bin_file="${matches[0]}"
+  sas_install_dir="${sas_bin_file%%-bin*}"
+fi
+###########################################################
+
+
+############# Setup the SAS Conda environment #############
+# Creates a Conda definition file that can be used to setup the environment that SAS
+#  will be associated with
+# We use the Conda-hosted HEASoft to avoid downloading and building HEASoft manually
+cat <<EOF > conda-sas.yml
+name: sas
+channels:
+  - https://heasarc.gsfc.nasa.gov/FTP/software/conda
+  - conda-forge
+dependencies:
+  - python=$py_version
+  - pytest
+  - xorg-libx11
+  - xorg-libxext
+  - xorg-libsm
+EOF
+
+# Use the yml to create the SAS env
+bash /usr/local/bin/conda-env-install.sh
+###########################################################
+
+
+############ Installing SAS Python requirements ###########
+# Use the Python requirements file included in the SAS directory to install
+#  the module that it wants for some included Python bits (e.g. pySAS)
+# We remove the pyDS9 requirement - the package is archived, and making it on Fornax would be a waste of time
+sed -i '/pyds9/d' sas_python_packages.txt
+micromamba run -n sas pip install -r sas_python_packages.txt --no-cache-dir
+###########################################################
+
+
+############ Moving unpacked SAS and installing ###########
+# Moves all of the files unpacked from the SAS download into the conda environment directory
+#  for SAS - it is easier to install SAS in-situ, rather than installing it then moving it, as
+#  some file paths get baked in during the installation process
+mv * $ENV_DIR/sas/
+# We must follow the unpacked files
+cd $ENV_DIR/sas/
+
+# Run the SAS install script, specifically in the environment we've just created
+micromamba run -n sas ./install.sh
+###########################################################
+
+
+################ Add conda (de)activation scripts ###############
+# Ensure that the directories we need actually exist
+mkdir -p $ENV_DIR/sas/etc/conda/activate.d
+mkdir -p $ENV_DIR/sas/etc/conda/deactivate.d
+
+# These scripts set up SAS and handles additional environment variable setting
+cat <<EOF > $ENV_DIR/sas/etc/conda/activate.d/sas-general_activate.sh
+#!/usr/bin/bash
+
+# SAS can be very particular about Perl - this is the path we set when SAS was 'built'
+export SAS_PERL=$SAS_PERL
+# And this is the conda environment we set up for it
+export SAS_PYTHON=$SAS_PYTHON
+
+# Adds the SAS conda environment library to the library path - without this
+#  we will get errors about not being able to find libsm.so.6
+export LD_LIBRARY_PATH="\$LD_LIBRARY_PATH:$ENV_DIR/sas/lib"
```

Comment:
Add `$ENV_DIR/heasoft/lib` too to remove the need for installing xorg-* packages above